float32 angleX
float32 angleY
float32 goalX
float32 goalY
float32 distance